daffodil-core/src/main/scala/org/apache/daffodil/dsom/AnnotatedSchemaComponent.scala

```diff
@@ -65,10 +65,21 @@ object ResolvesProperties {
     "outputTypeCalc",
     "repType",
     "repValueRanges",
-    "repValues"
-  )
+    "repValues")
 }
 
+// TODO: ResolvesProperties should really only be mixed-in by Term.
+// It is very easy to make the mistake of looking up properties on something like
+// a global def/decl, and then you aren't getting the combining of that with
+// the referencing object and its default format (from its containing schema document).
+//
+// We need to refactor and provide ResolvesScopedProperties, and ResolvesLocalProperties
+// mixins so that non-terms can use different method names (but similar idioms) to get local
+// properties. That way lookup of properties that is supposed to be scoped can't be accidently
+// called on non-terms. We really do not want OO-style polymorphism on property finding
+// across terms and non-terms. Only for the terms. Non-terms want their own local-only
+// method names.
```

Comment:
Should we create a ticket for this?

Reply:
There are quite a lot of cleanups that are listed as TODO in the code. DAFFODIL-1536 is the ticket for cleaning up all the TODOs and FIXMEs in the code. 

This one is an improvement to maintainability and would make the classes enforce the usage patterns better, but I'm disinclined to mess with this until DSOM stabilizes (the backpointers thing), or fix it as part of that.